### PR TITLE
Show level details dialog on lesson plans

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Launch the level details dialog! Some levels are still a bit rough around the edges and will be addressed in a polish focused pass, but I want to get this out the door so that curriculum writers can see it!

Screen capture (the "Help & Tips" tab can be slow to switch to links but I don't believe that's a bug with the dialog)
https://user-images.githubusercontent.com/46464143/112197551-7276ae80-8bc9-11eb-8202-6a8b35dfe840.mov

This PR also makes some improvements to match levels to show more of the question. This also shows one of the biggest gaps between CB pullthrough and this dialog for certain level types, which is that it doesn't always show the teacher markdown. I'll look at that in a future PR
Screen Shot 2021-03-23 at 11 05 39 AM
vs on CB
Screen Shot 2021-03-23 at 11 06 21 AM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by dmcavoy** [comment]
> Screen capture (the "Help & Tips" tab can be slow to switch to links but I don't believe that's a bug with the dialog)

This was already true in the levels

**Comment by dmcavoy** [comment]
> This also shows one of the biggest gaps between CB pullthrough and this dialog for certain level types, which is that it doesn't always show the teacher markdown. I'll look at that in a future PR

You may already know this but my guess is its because the contained level hold the instructions but the container level has the teacher markdown

Edit: I was guessing that based on the text of the question. If its not a contained level then this is probably not the issue. My other guess would be the DSL saves teacher markdown in a different way than other level types.

File: apps/test/unit/templates/lessonOverview/activities/ProgressionDetailsTest.jsx [hunk 1]
```diff
@@ -16,4 +16,10 @@ describe('ProgressionDetails', () => {
     const wrapper = shallow(<ProgressionDetails {...defaultProps} />);
     expect(wrapper.find('ProgressLevelSet').length).to.equal(1);
   });
+
+  it('can show level details dialog after bubble click', () => {
+    const wrapper = shallow(<ProgressionDetails {...defaultProps} />);
+    wrapper.instance().handleBubbleClick({id: 1});
+    expect(wrapper.find('Connect(LevelDetailsDialog)').length).to.equal(1);
```
Comment (dmcavoy) [inline apps/test/unit/templates/lessonOverview/activities/ProgressionDetailsTest.jsx:24]: Maybe check that its length is zero before the click?

**Comment by dmcavoy** [comment]
Screen Shot 2021-03-24 at 12 37 53 PM

Not blocking. Looks like the tabs turn white when you hover. I don't think thats how they look in the level.

**Comment by dmcavoy** [comment]
Also not blocking. The immersive reader button is giving errors.

**Review by dmcavoy — APPROVED** [review]
Left some comments on things I noticed as I was testing but I think its ready to ship. Following up with a polish pass after getting it in front of curriculum makes sense to me.

**Comment by bethanyaconnor** [comment]
> Edit: I was guessing that based on the text of the question. If its not a contained level then this is probably not the issue. 

I haven't actually checked for contained level, but the screenshot was from a Multi level. I think those just put the blue teacher markdown box on the page so I'll probably do that

> The immersive reader button is giving errors.

It gives errors for me on level pages too so I assumed I don't have a token. It's actually one the reasons I'd like to see this live so I can test this kind of thing

> Looks like the tabs turn white when you hover. I don't think thats how they look in the level.

Hmm yeah you're right. I'll take a quick look to see if it's an easy fix and if not I'll fix in a future PR.

**Comment by bethanyaconnor** [comment]
Okay, I found the issue with the tab hover -- we have some CSS in `common.scss` that overrides some styles in `application.scss`. I didn't want to include the entirety of `common.scss` here, so I just added a small style rule to `lessons.scss`. I'm pretty unfamiliar about the conventions around this though so let me know if I should've done something else!

**Comment by davidsbailey** [comment]
> Okay, I found the issue with the tab hover -- we have some CSS in `common.scss` that overrides some styles in `application.scss`. I didn't want to include the entirety of `common.scss` here, so I just added a small style rule to `lessons.scss`. I'm pretty unfamiliar about the conventions around this though so let me know if I should've done something else!

@bethanyaconnor just trying to follow along... which CSS in `common.scss`?

**Comment by bethanyaconnor** [comment]
@davidsbailey The CSS [here](https://github.com/code-dot-org/code-dot-org/blob/36b2aba9dfbd9145c5d61b29e025f77b1af214c9/apps/style/common.scss#L64)

**Pull request merged**